src/core/CustomData.cpp

```diff
@@ -125,15 +152,41 @@ void CustomData::copyDataFrom(const CustomData* other)
     emitModified();
 }
 
-QDateTime CustomData::getLastModified() const
+QDateTime CustomData::lastModified() const
 {
     if (m_data.contains(LastModified)) {
-        return Clock::parse(m_data.value(LastModified));
+        return Clock::parse(m_data.value(LastModified).value);
+    }
+
+    // Try to find the latest modification time in items as a fallback
+    QDateTime modified;
+    for (auto i = m_data.constBegin(); i != m_data.constEnd(); ++i) {
+        if (i->lastModified.isValid() && (!modified.isValid() || i->lastModified > modified)) {
+            modified = i->lastModified;
+        }
     }
-    return {};
+    return modified;
+}
+
+QDateTime CustomData::lastModified(const QString& key) const
+{
+    return m_data.value(key).lastModified;
 }
 
-bool CustomData::isProtectedCustomData(const QString& key) const
+void CustomData::updateLastModified(QDateTime lastModified)
+{
+    if (m_data.isEmpty() || (m_data.size() == 1 && m_data.contains(LastModified))) {
+        m_data.remove(LastModified);
+        return;
+    }
+
+    if (!lastModified.isValid()) {
+        lastModified = Clock::currentDateTimeUtc();
+    }
+    m_data.insert(LastModified, {lastModified.toString(), QDateTime()});
+}
```

Comment:
Do we want to keep doing this now that each item has a modified timestamp? My thought would be that this would be replaced on load with the modified timestamps:
1. Database read from file into memory
2. Database custom data checks for _LAST_MODIFIED entry
3. If present, set all last modified attributes of other custom data elements and delete _LAST_MODIFIED
4. On Save, this new state is set (no _LAST_MODIFIED) and the database is upgraded to KDBX 4.1

We shouldn't have _LAST_MODIFIED items in entry and group custom data since the date is derived from the entry/group modified date.

Reply:
I described that in the commit message. I left the old behaviour untouched, because otherwise CustomData became unmergeable due to the lack of DeletedObjects. This thing here is ugly as hell, but I don't see a good way to avoid it right now. Having dates on CustomData is great, but useless as of now, since you cannot detect deletions. My implementation effectively only prevents that KeePassXC discards dates set in the file, but doesn't really make use of them so far.